crates/oxc_linter/src/rules/unicorn/prefer_default_parameters.rs

```diff
@@ -0,0 +1,535 @@
+    first_stmt.span() == node.kind().span()
+}
+
+fn check_no_extra_references<'a>(
+    ctx: &LintContext<'a>,
+    param_ident_span: Span,
+    param: &FormalParameter<'a>,
+) -> bool {
+    // Get the symbol for the parameter
+    let BindingPatternKind::BindingIdentifier(binding_ident) = &param.pattern.kind else {
+        return false;
+    };
+
+    let Some(symbol_id) = binding_ident.symbol_id.get() else {
+        return false;
+    };
+
+    // Check how many times the parameter is referenced
+    let references: Vec<_> = ctx.scoping().get_resolved_references(symbol_id).collect();
+
+    // For `const bar = foo || 'bar'`, the parameter should only be referenced once
+    // (in the LogicalExpression)
+    if references.len() != 1 {
+        return false;
+    }
+
+    // The single reference should be the one in the LogicalExpression
+    let reference = &references[0];
+    ctx.semantic().reference_span(reference) == param_ident_span
+}
+
+fn check_no_extra_references_assignment<'a>(
+    ctx: &LintContext<'a>,
+    param_ident_span: Span,
+    param: &FormalParameter<'a>,
+) -> bool {
+    // Get the symbol for the parameter
+    let BindingPatternKind::BindingIdentifier(binding_ident) = &param.pattern.kind else {
+        return false;
+    };
+
+    let Some(symbol_id) = binding_ident.symbol_id.get() else {
+        return false;
+    };
+
+    // For assignment `foo = foo || 'bar'`, the parameter should have exactly 2 references:
+    // 1. The read in the LogicalExpression (foo || 'bar')
+    // 2. The write in the AssignmentExpression (foo = ...)
+    let references: Vec<_> = ctx.scoping().get_resolved_references(symbol_id).collect();
+
+    if references.len() != 2 {
+        return false;
+    }
+
+    // One should be a read (in logical expr), one should be a write (assignment target)
+    let reads: Vec<_> = references.iter().filter(|r| !r.is_write()).collect();
+    let write_count = references.iter().filter(|r| r.is_write()).count();
+
+    if reads.len() != 1 || write_count != 1 {
+        return false;
+    }
+
+    // The read should be at the same span as param_ident
+    ctx.semantic().reference_span(reads[0]) == param_ident_span
+}
+
+fn check_expression<'a>(
+    ctx: &LintContext<'a>,
+    node: &AstNode<'a>,
+    left_name: &str,
+    right: &Expression<'a>,
+    is_assignment: bool,
+    stmt_span: Span,
+) {
+    // Right must be a LogicalExpression with || or ??
+    let Expression::LogicalExpression(logical_expr) = right.without_parentheses() else {
+        return;
+    };
+
+    if !matches!(logical_expr.operator, LogicalOperator::Or | LogicalOperator::Coalesce) {
+        return;
+    }
+
+    // Left side of logical expression must be an identifier
+    let Expression::Identifier(param_ident) = logical_expr.left.without_parentheses() else {
+        return;
+    };
+
+    let param_name = param_ident.name.as_str();
+
+    // Right side of logical expression must be a literal
```

Comment:
most of these comments are unnecessary, as the code already explains what it's doing